python_modules/dagster/dagster/_utils/container.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import os
+from typing import Optional, TypedDict
+
+import pendulum
+
+
+def cpu_usage_path():
+    return os.getenv("DAGSTER_CPU_USAGE_PATH", "/sys/fs/cgroup/cpuacct/cpuacct.usage")
+
+
+def cpu_info_path():
+    return os.getenv("DAGSTER_CPU_INFO_PATH", "/proc/cpuinfo")
+
+
+def memory_usage_path():
+    return os.getenv("DAGSTER_MEMORY_USAGE_PATH", "/sys/fs/cgroup/memory/memory.usage_in_bytes")
+
+
+def memory_limit_path():
+    return os.getenv("DAGSTER_MEMORY_LIMIT_PATH", "/sys/fs/cgroup/memory/memory.limit_in_bytes")
+
+
+class UtilizationMetrics(TypedDict):
+    num_allocated_cores: Optional[int]
+    cpu_time: Optional[float]
+    cpu_utilization: Optional[float]
+    memory_utilization: Optional[float]
+    memory_limit: Optional[int]
+    measurement_timestamp: float
+
+
+def retrieve_containerized_utilization_metrics(
+    logger: Optional[logging.Logger],
+    last_cpu_measurement_time: Optional[float],
+    last_cpu_measurement_value: Optional[float],
+) -> UtilizationMetrics:
+    """Retrieve the CPU and memory utilization metrics from cgroup and proc files."""
+    measurement_timestamp = pendulum.now("UTC").timestamp()
+    cpu_time = _retrieve_containerized_cpu_time(logger)
+    num_cores = _retrieve_containerized_num_allocated_cores(logger)
+    if cpu_time and last_cpu_measurement_time and last_cpu_measurement_value and num_cores:
+        cpu_utilization = (
+            (cpu_time - last_cpu_measurement_value)
+            / (measurement_timestamp - last_cpu_measurement_time)
+            / num_cores
```

Comment:
Can we be consistent on the unit variable names here? Hard to tell if `last_cpu_measurement_time` is referring to elapsed time or it's another CPU time.